File: fixtures/scale/r/medium/core/history.r
# In-memory expression history.

.calc_history <- new.env()
.calc_history$entries <- character(0)

history_add <- function(expr, result) {
  .calc_history$entries <- c(.calc_history$entries, sprintf("%s = %g", expr, result))
}

history_show <- function() {
  for (entry in .calc_history$entries) {
    cat(entry, "\n")
  }
}

history_clear <- function() {
  .calc_history$entries <- character(0)
}

File: fixtures/scale/julia/medium/src/History.jl
module History

export history_add, history_entries, history_clear

const ENTRIES = String[]

function history_add(expr::String, result::Float64)
    push!(ENTRIES, string(expr, " = ", result))
end

function history_entries()
    return copy(ENTRIES)
end

function history_clear()
    empty!(ENTRIES)
end

end
